### PR TITLE
Fixes null != undefined error in homePagePreferenceSpec.js

### PR DESCRIPTION
**Pull request by benjaminapetersen**
I tried this with a few versions of node & npm:

```
# first
node -v 6.10.2; npm -v 3.10.10
# then
node -v 9.4.0; npm -v 5.6.0
```

But either way the test failed due to `null != undefined`.  This tiny change gets the test passing again, though I'm not sure it actually fixes the test.  `getHomePagePath - should notify invalid project` is confusing to me in light of the `expect` statement.

Thoughts @dtaylor113 ?

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
I'm a little bit inclined towards updating the `describe()` and `it()` to be more like:

```javascript 
describe('#getHomePagePath', function() {
  describe('when given an empty config object', function() {
    it('should set the default home page to /catalog', function() {
        // expect
    });
  });

   // etc.
});
```
It reads more clearly & makes it easier to fix a test when you aren't familiar with the domain.

**Comment by dtaylor113** [comment]
> getHomePagePath - should notify invalid project is confusing to me in light of the expect statement.

When you have an invalid (non-existent) project as a home page preference, notifyInvalidProjectHomePage(..) (a) Clears the homePagePref local storage, then (b) pops up a warning toast notification, so the `expect` in the unit test is trying to verify that the local storage/home page pref has been cleared/removed.

File: test/spec/services/homePagePreferenceSpec.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ describe("HomePagePreferenceService", function(){
       };
       HomePagePreferenceService.setHomePagePreference(homePagePref);
       HomePagePreferenceService.notifyInvalidProjectHomePage(homePagePref.project);
-      expect(localStorage.getItem("openshift/home-page-pref/")).not.toBeDefined();
+      expect(localStorage.getItem("openshift/home-page-pref/")).toBeFalsy();
```
Comment (spadgett) [inline test/spec/services/homePagePreferenceSpec.js]: I'm unclear why this should be necessary since we spyOn localStorage.removeItem to delete the key. It should always be undefined?

https://github.com/openshift/origin-web-console/pull/2736/files#diff-2e146cc351b88485c09870a5cc2f86f9R23

https://github.com/openshift/origin-web-console/blob/4e1c80cf375170e976c77ee8910d0aafacf62e16/app/scripts/services/homePagePreference.js#L38-L44

**Comment by benjaminapetersen** [comment]
@dtaylor113 but the unit test doesn't test the notification, so this language is misleading in this context.  I think the notification part would make sense in an e2e test, or in another test for the page itself that verifies the notification appears when the service returns `null`/`undefined`.

File: test/spec/services/homePagePreferenceSpec.js (hunk 1 above)
Comment (benjaminapetersen) [reply test/spec/services/homePagePreferenceSpec.js]: @dtaylor113  thoughts?

@spadgett good Q, I'm only aiming to remove the failing test blocker with this PR.

File: test/spec/services/homePagePreferenceSpec.js [hunk 2]
```diff
@@ -8,20 +8,20 @@ describe("HomePagePreferenceService", function(){
     inject(function(_HomePagePreferenceService_){
       HomePagePreferenceService = _HomePagePreferenceService_;
     });
-    
+
```
Comment (benjaminapetersen) [inline test/spec/services/homePagePreferenceSpec.js]: We should also double check that everyone is using the same `.editorconfig` so we have consistent white space settings.

Comment (benjaminapetersen) [reply test/spec/services/homePagePreferenceSpec.js]: (just to avoid unnecessary diffs)

**Comment by spadgett** [comment]
@benjaminapetersen I agree the test description isn't quite right

File: test/spec/services/homePagePreferenceSpec.js (hunk 2 above)
Comment (spadgett) [reply test/spec/services/homePagePreferenceSpec.js]: @yapei FYI, it would be good to set up editorconfig for your editor

http://editorconfig.org/

File: test/spec/services/homePagePreferenceSpec.js (hunk 1 above)
Comment (spadgett) [reply test/spec/services/homePagePreferenceSpec.js]: I'm able to reproduce locally, but I worry it means `spyOn` is not working. Strange we don't see this in travis or jenkins. Maybe it's a chrome vs firefox thing.

**Comment by benjaminapetersen** [comment]
@spadgett @dtaylor113 my own tests burned me on the last updates to `Membership`, so I'd really like to see our wording be as descriptive & specific as possible with each unit test. Granted, I think this service is probably small enough it's not going to cause us a ton of pain, but we can set the trajectory for future more complex tests.

**Comment by spadgett** [comment]
@benjaminapetersen FYI @dtaylor113 didn't write the original test in case you weren't aware :)

**Comment by benjaminapetersen** [comment]
Ha oh? I thought he owned this feature so I've been pinging him.  Oops, my bad 😄

**Comment by dtaylor113** [comment]
> Ha oh? I thought he owned this feature so I've been pinging him. Oops, my bad

lol..no worries @benjaminapetersen

**Comment by benjaminapetersen** [comment]
@spadgett yup, ill look into the spy issue.

It is very strange that this works in travis/jenkins but not locally.

**Comment by dtaylor113** [comment]
> It is very strange that this works in travis/jenkins but not locally.

The unit tests work for me locally, Fedora 27, PhantomJS 2.1.1

**Comment by spadgett** [comment]
I think it is a difference in native behavior of removeItem. If spyOn is not working, we can have different behavior across browser

File: test/spec/services/homePagePreferenceSpec.js (hunk 2 above)
Comment (benjaminapetersen) [reply test/spec/services/homePagePreferenceSpec.js]: @yapei & thanks for taking the time to add these tests!  Quirk with `jasmine` aside, every little bit helps!

File: test/spec/services/homePagePreferenceSpec.js (hunk 1 above)
Comment (benjaminapetersen) [reply test/spec/services/homePagePreferenceSpec.js]: Works locally if I switch the browser to Chrome.
It's a reported bug that spyOn localStorage doesn't work in Firefox.  Figuring out a workaround.

Comment (benjaminapetersen) [reply test/spec/services/homePagePreferenceSpec.js]: `spyOn` wont work, mocking `window.localStorage` or `$window.localStorage` also does not work as `window.localStorage`as a data property as not writable and not configurable. 

Neat 😄

Comment (spadgett) [reply test/spec/services/homePagePreferenceSpec.js]: OK I am good with the change if spyOn is working on other browsers and it makes the tests pass on Firefox.

File: test/spec/services/homePagePreferenceSpec.js (hunk 2 above)
Comment (yapei) [reply test/spec/services/homePagePreferenceSpec.js]: @benjaminapetersen  @spadgett  I'm really sorry for the inconsistency in my PR, I should setup editorconfig before opening a PR as documented in our repo [before opening a PR](https://github.com/openshift/origin-web-console#before-opening-a-pull-request)

**Comment by yapei** [comment]
> getHomePagePath - should notify invalid project is confusing to me in light of the expect statement.
@benjaminapetersen  Thanks for pointing it out. I will update descriptions, it's my fault giving unclear statement.

>  I'm a little bit inclined towards updating the describe() and it() to be more like:
@benjaminapetersen  @dtaylor113  Thanks for your comments, will update also.

File: test/spec/services/homePagePreferenceSpec.js (hunk 2 above)
Comment (benjaminapetersen) [reply test/spec/services/homePagePreferenceSpec.js]: No worries, not a big deal.  Appreciate it!

**Comment by yapei** [comment]
@benjaminapetersen  Actually I saw you already have an PR #2741 updating them, thank you so much! I would pay more attention to the language/description in my following PR.

**Comment by benjaminapetersen** [comment]
Yup we got it @yapei, thanks again! Def appreciate the added tests, we have plenty of room for more. 😄
